psyneulink/core/compositions/composition.py

```diff
@@ -10057,12 +10227,16 @@ def _update_learning_parameters(self, context):
         pass
 
     @handle_external_context(execution_id=NotImplemented)
-    def reset(self, values, context=NotImplemented):
-        if context.execution_id is NotImplemented:
-            context.execution_id = self.most_recent_context.execution_id
+    def reset(self, values=None, context=NotImplemented):
+        if context is NotImplemented:
+            context = self.most_recent_context
 
-        for i in range(self.stateful_nodes):
-            self.stateful_nodes[i].reset(values[i], context=context)
+        if not values:
+            values = {}
+
+        for node in self.stateful_nodes:
+            reset_val = None if not node in values else values[node]
```

Comment:
same here (`None` is the default reutnred by `get`)
`reset_val = values.get(node)`

Reply:
updated in ffad3efb0747fedd8306bad83a2bfc5dd5ddee29